### ISSUE TITLE
[JENKINS-4186] Builds triggered on startup although they have not changed

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#7618]
<p>After upgrade from 1.318 to 1.319, Hudson decided that (probably) all of our<br/>
projects should be rebuilt because of SCM changes. However, none of them had any<br/>
changes.<br/>
This message in my "Perforce Polling Log" causes me to file the issue under the<br/>
perforce plugin - but I don't know if it's just here?</p>

<p>Workspace is offline.</p>

<p>Scheduling a new build to get a workspace.</p>

<p>Done. Took 0 ms</p>

<p>Changes found</p>

<p>Issue 4182 also concerns workspace problems in hudson 1.319, so perhaps it's a<br/>
core issue instead?</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=torbent">torbent</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-4186" target="_blank">Builds triggered on startup although they have not changed</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2011-07-10T13:34:30+00:00
<li><b>votes</b>: 2
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>After upgrade from 1.318 to 1.319, Hudson decided that (probably) all of our
projects should be rebuilt because of SCM changes. However, none of them had any
changes.
This message in my "Perforce Polling Log" causes me to file the issue under the
perforce plugin - but I don't know if it's just here?</p>

<p>Workspace is offline.</p>

<p>Scheduling a new build to get a workspace.</p>

<p>Done. Took 0 ms</p>

<p>Changes found</p>

<p>Issue 4182 also concerns workspace problems in hudson 1.319, so perhaps it's a
core issue instead?</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: All, OS: All</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torbent's <a href="https://issues.jenkins.io/browse/JENKINS-4186?focusedId=131998&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-131998">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Moved to 'core' subcomponent because it appears to not be a 'perforce' problem.</p>
</pre>
</details>

</details>
<p>Moved to 'core' subcomponent because it appears to not be a 'perforce' problem.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke's <a href="https://issues.jenkins.io/browse/JENKINS-4186?focusedId=132000&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-132000">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>This is due to the internal change in Hudson 1.319 about how a workspace is managed.</p>

<p>This should happen only once after upgrade, and while I did think about if
there's any way to avoid this, I couldn't. This affects all jobs that have SCM
polling set up, regardless of SCM.</p>

<p>I'm marking this as WONTFIX, although I'm ashamed of it.</p>
</pre>
</details>

</details>
<p>This is due to the internal change in Hudson 1.319 about how a workspace is managed.</p>

<p>This should happen only once after upgrade, and while I did think about if<br/>
there's any way to avoid this, I couldn't. This affects all jobs that have SCM<br/>
polling set up, regardless of SCM.</p>

<p>I'm marking this as WONTFIX, although I'm ashamed of it.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>bhalter's <a href="https://issues.jenkins.io/browse/JENKINS-4186?focusedId=139942&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-139942">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Any chance of getting a second look at this?  </p>

<p>I'm trying to upgrade from 1.313 to the current release and stumbling over this now.  I've got about 500 builds that are managed by Hudson and having each of them need a rebuild is a particularly ugly solution for us.  Given that we need to script up work-arounds for some changes in the config.xml and to enable some new features I'm not opposed to scripting up a work around for this.  </p>

<p>Can you be more descriptive of the change to workspaces in 1.319?</p>
</pre>
</details>

</details>
<p>Any chance of getting a second look at this?  </p>

<p>I'm trying to upgrade from 1.313 to the current release and stumbling over this now.  I've got about 500 builds that are managed by Hudson and having each of them need a rebuild is a particularly ugly solution for us.  Given that we need to script up work-arounds for some changes in the config.xml and to enable some new features I'm not opposed to scripting up a work around for this.  </p>

<p>Can you be more descriptive of the change to workspaces in 1.319?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/evernat.png" title="evernat's avatar" /> evernat's <a href="https://issues.jenkins.io/browse/JENKINS-4186?focusedId=150330&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-150330">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Sorry bhalter but I suppose that your 500 jobs are built now.
So closing again as won't fix.</p>
</pre>
</details>

</details>
<p>Sorry bhalter but I suppose that your 500 jobs are built now.<br/>
So closing again as won't fix.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torbent's <a href="https://issues.jenkins.io/browse/JENKINS-4186?focusedId=157285&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-157285">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>It turned out to be an "only-once" problem. Should have been closed ages ago.</p>
</pre>
</details>

</details>
<p>It turned out to be an "only-once" problem. Should have been closed ages ago.</p>
